### ISSUE TITLE
Cannot revoke permit of `MintBurnToken`

### ISSUE THREAD

**Issue by MarioPoneder** [shieldify-security/Portals-810fd90645cccf6fd80d91f8050eff618f63af81#23]
## Severity
Low-Medium

## Description
The `MintBurnToken` inherits from `ERC20Permit` which provides users with a gasless way to give other users a token approval by generating a permit signature.  
However, this signature is valid until the `deadline` and cannot be revoked beforehand if desired.

## Impact
The `MintBurnToken` does not implement a pulic entry-point for `ERC20Permit._useNonce(...)` which could be used to invalidate a token approval signature by incrementing the nonce.

## Attack Scenario
-

## Proof of Concept (PoC)
-

## Location of Affected Code

https://github.com/shieldify-security/Portals-810fd90645cccf6fd80d91f8050eff618f63af81/blob/30129e4ce71a379e0bc62a93c245e5898331028e/src/V2MultiAsset/MintBurnToken.sol#L10

## Recommendation

Add a public entry-point of `_useNonce(...)` to the  `MintBurnToken`:
```solidity
function useNonce(address owner) external returns (uint256 current) {
    current = _useNonce(owner);
}
```

**Comment by PossumLabsCrypto** [comment]
Is this not a vulnerability of the permit library itself? Why did nobody tell OpenZeppelin before releasing this stuff to the public? 👀 

I think I need a second opinion here to determine where responsibilities lie. If as a developer, I´m supposed to check/audit every library I use from renown firms like OpenZeppelin, I might as well just write them myself in the first place. lol

@ShieldifyMartin @Ptodoroff @anontheauditor How would you handle this?

**Comment by carrotsmuggler** [comment]
I am not the submitter of this issue, however I have come across this discussion before on contests on other platforms and I hope my input helps. 

OZ does not consider this a bug, because a user can use up a nonce by just calling `permit` again with any valid signed message. This has pretty much the same complexity from the user's side as calling some `useNonce` function, since both do the same thing (update nonce) in one transaction and since the user already signed a permit message, it is assumed they can just sign another one.

**Comment by MarioPoneder** [comment]
I agree, it's a valid workaround to create another valid signature with the same "nonce" and use it via `permit()` in order to invalidate the previous signature. But an unnecessarily lengthy process when a signature needs to be invalidated quickly.

Anyways, I just wanted @PossumLabsCrypto to be aware of that, the final severity assessment and mitigation steps are fully up to you.

**Comment by PossumLabsCrypto** [comment]
Thanks for the input!

Imo the reality is that most users don´t interact directly with contracts and increasingly so in the future as we (the industry) onboard more normies over time. 
Hence, the UX improvement is mainly for people who know their way around anyways. 🤔 

I acknowledge it as low this time and will evaluate if it´s worth fixing.
